src/sentry/seer/similarity/utils.py

```diff
@@ -224,18 +225,14 @@ def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
         frame_count += len(contributing_frames)
 
         for frame in contributing_frames:
-            frame_dict = {"filename": "", "function": "", "context-line": "", "module": ""}
-            for frame_values in frame.get("values", []):
-                if frame_values.get("id") in frame_dict:
-                    frame_dict[frame_values["id"]] = _get_value_if_exists(frame_values)
+            frame_dict = extract_values_from_frame_values(frame.get("values", []))
+            filename = extract_filename(frame_dict)
 
             if not _is_snipped_context_line(frame_dict["context-line"]):
                 found_non_snipped_context_line = True
 
-            if frame_dict["filename"] == "" and frame_dict["module"] == "":
+            if not filename:
                 has_no_filename_or_module = True
-            elif frame_dict["filename"] == "":
-                frame_dict["filename"] = frame_dict["module"]
```

Comment:
This is now handled within `extract_filename`:
<img width="331" alt="image" src="https://github.com/user-attachments/assets/fe5d013b-5d4f-4b24-990b-b0c34fa42801">